src/Serilog.Sinks.File/Sinks/File/WriteCountingStream.cs

```
@@ -60,7 +60,7 @@ public override long Position
 
         public override long Seek(long offset, SeekOrigin origin)
         {
-            throw new NotSupportedException();
+            return _stream.Seek(offset, origin);
```

Would the workaround be impacted if we continue to throw here? Although it appears to break the `CanSeek`/`Seek()` contract, actually calling `Seek()` would break the write counting. Perhaps as a compromise we could throw `InvalidOperationException()` instead?